cupy/_indexing/generate.py

```diff
@@ -69,7 +69,7 @@ def __getitem__(self, key):
 
         final_dtype = numpy.result_type(*arrays, *scalars)
         if final_dtype is not None:
-            for k in scalars:
+            for k in range(len(scalars)):
```

Comment:
`k` should be indices of scalars in `objs`. How about the following one?

```py
        final_dtype = numpy.result_type(*arrays, *[objs[k] for k in scalars])
        if final_dtype is not None:
            for k in scalars:
                objs[k] = objs[k].astype(final_dtype)
```

Reply:
We cannot use the scalars as indices since they would not be integers, another alternative would be to store the scalar positions into another list and then use that one

Reply:
Yes, exactly. So how about revert line 59 to `scalars.append(i)` and fix L70-L73 as I suggested?